iio/iio_app/iio_app.c

```diff
@@ -79,31 +79,11 @@
 #define UART_BAUDRATE_DEFAULT	115200
 #define UART_STOPBITS_DEFAULT	NO_OS_UART_STOP_1_BIT
 
-static inline uint32_t _calc_uart_xfer_time(uint32_t len, uint32_t baudrate)
-{
-	uint32_t ms = 1000ul * len * 8 / UART_BAUDRATE_DEFAULT;
-	ms += ms / 10; // overhead
-	return ms;
-}
-
 #if !defined(LINUX_PLATFORM) && !defined(NO_OS_NETWORKING) && !defined(NO_OS_USB_UART)
 static int32_t iio_print_uart_info_message(struct no_os_uart_desc **uart_desc,
-		struct no_os_uart_init_param *user_uart_params,
 		char *message, int32_t msglen)
 {
-	int32_t status;
-	uint32_t delay_ms;
-
-	status = no_os_uart_write(*uart_desc, (uint8_t *)message, msglen);
-	if (status < 0)
-		return status;
-
-	delay_ms = _calc_uart_xfer_time(msglen, UART_BAUDRATE_DEFAULT);
-	no_os_mdelay(delay_ms);
-
-	/** Reinitialize UART with parameters required by the IIO application */
-	no_os_uart_remove(*uart_desc);
```

Comment:
IIRC this was needed and is still needed because there is no API to change baudrate once the UART was initialized. And sometimes the iiod runs at higher baudrate than the iio greeting message. I think adding an UART baudrate change API would be the proper way to address this mess.

And because there was no API, we had to "needlessly" _remove() and _init() with a delay in between because on fast processors or slow UARTs or both, you'd see a partial iio greeting message.

Reply:
so, if that's the case then i'd need to re-visit this

because the `no_os_uart_remove()` + `no_os_uart_init()` doesn't seem to initialize properly;
some init_data gets lost